[Bug]: Error shown in the logs when stopping Hop GUI

### Apache Hop version?

2.19.0-SNAPSHOT

### Java version?

OpenJDK 21

### Operating system

Linux

### What happened?

This error is shown on the console when simply exiting the Hop GUI:

```
Serious error detected in the Hop GUI: Failed to execute runnable (org.eclipse.swt.SWTException: Widget is disposed)
org.eclipse.swt.SWTException: Failed to execute runnable (org.eclipse.swt.SWTException: Widget is disposed)
        at org.eclipse.swt.SWT.error(SWT.java:4950)
        at org.eclipse.swt.SWT.error(SWT.java:4865)
        at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:135)
        at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:5084)
        at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:4549)
        at org.eclipse.swt.widgets.Display.release(Display.java:4617)
        at org.eclipse.swt.graphics.Device.dispose(Device.java:297)
        at org.apache.hop.ui.hopgui.HopGui.open(HopGui.java:632)
        at org.apache.hop.ui.hopgui.HopGui.main(HopGui.java:453)
        at org.apache.hop.ui.hopgui.HopCommandGui.run(HopCommandGui.java:57)
        at picocli.CommandLine.executeUserObject(CommandLine.java:2045)
        at picocli.CommandLine.access$1500(CommandLine.java:148)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2469)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2461)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2423)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2277)
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2425)
        at picocli.CommandLine.execute(CommandLine.java:2174)
        at org.apache.hop.hop.Hop.main(Hop.java:131)
Caused by: org.eclipse.swt.SWTException: Widget is disposed
        at org.eclipse.swt.SWT.error(SWT.java:4950)
        at org.eclipse.swt.SWT.error(SWT.java:4865)
        at org.eclipse.swt.SWT.error(SWT.java:4836)
        at org.eclipse.swt.widgets.Widget.error(Widget.java:598)
        at org.eclipse.swt.widgets.Widget.checkWidget(Widget.java:513)
        at org.eclipse.swt.widgets.MenuItem.getEnabled(MenuItem.java:517)
        at org.eclipse.swt.widgets.MenuItem.isEnabled(MenuItem.java:713)
        at org.apache.hop.ui.core.gui.GuiMenuWidgets.enableMenuItem(GuiMenuWidgets.java:368)
        at org.apache.hop.ui.hopgui.HopGui.handleFileCapabilities(HopGui.java:1913)
        at org.apache.hop.ui.hopgui.HopGui.handleFileCapabilities(HopGui.java:1898)
        at org.apache.hop.ui.hopgui.file.pipeline.HopGuiPipelineGraph.lambda$updateGui$34(HopGuiPipelineGraph.java:6024)
        at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:40)
        at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:132)
        ... 16 more

```


### Issue Priority

Priority: 3

### Issue Component

Component: Hop Gui

--- a/ui/src/main/java/org/apache/hop/ui/core/gui/GuiMenuWidgets.java
+++ b/ui/src/main/java/org/apache/hop/ui/core/gui/GuiMenuWidgets.java
@@ -365,7 +365,7 @@ public class GuiMenuWidgets extends BaseGuiWidgets {
     boolean hasCapability =
         handler != null ? handler.hasCapability(permission) : fileType.hasCapability(permission);
     boolean enable = hasCapability && active;
-    if (menuItem != null && enable != menuItem.isEnabled()) {
+    if (menuItem != null && !menuItem.isDisposed() && enable != menuItem.isEnabled()) {
       menuItem.setEnabled(enable);
     }
     menuEnabledMap.put(id, enable);

--- a/ui/src/test/java/org/apache/hop/ui/core/gui/GuiMenuWidgetsDisposedTest.java
+++ b/ui/src/test/java/org/apache/hop/ui/core/gui/GuiMenuWidgetsDisposedTest.java
@@ -1,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hop.ui.core.gui;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.Map;
+import org.apache.hop.ui.hopgui.file.IHopFileType;
+import org.eclipse.swt.widgets.MenuItem;
+import org.junit.jupiter.api.Test;
+
+class GuiMenuWidgetsDisposedTest {
+
+  @Test
+  void ignoresDisposedMenuItemWhenUpdatingCapability() {
+    String id = "file-save";
+    String permission = "save";
+    IHopFileType fileType = mock(IHopFileType.class);
+    MenuItem menuItem = mock(MenuItem.class);
+    when(fileType.hasCapability(permission)).thenReturn(true);
+    when(menuItem.isDisposed()).thenReturn(true);
+
+    GuiMenuWidgets widgets = new GuiMenuWidgets();
+    widgets.setMenuItemMap(Map.of(id, menuItem));
+
+    widgets.enableMenuItem(fileType, id, permission);
+
+    verify(menuItem, never()).isEnabled();
+    verify(menuItem, never()).setEnabled(true);
+  }
+}

